### DISCUSSION TITLE
(merged with #1914) Bedtime mode

### DISCUSSION THREAD

**Discussion by smaybius** [InfiniTimeOrg/InfiniTime#1899] [Ideas]
During bedtime mode, notifications should be silent, the screen always the dimmest, bluetooth turned off, and any sleep tracking software doing work entirely on the device until returning data to the connected host the next morning.

**Comment by tausen** [comment]
Hmm, this kind of switching some settings depending on another brings a few tricky decisions:

- The user sets (for example) the screen brightness to medium
- The user enables "night mode", which in turn sets the brightness to minimum
- The user changes brightness to maximum 
- The user disables night mode -- should the brightness now stay at maximum because it was overwritten by the user, or should it return to medium because that's what it was before entering night mode?

Sure, we can probably come up with some rules -- but it might produce some sudden changes of settings that could seen unintuitive to the user. I could imagine bug reports that "switching to night mode makes watch loose bluetooth connection", for example. 

Hard decisions on what "night mode" means could also impose some undesired restrictions on how the watch is used?

Perhaps the night and do not disturb modes should be some kind of profiles with persistent configuration? So switching to night mode will set the brightness and bluetooth enablement to whatever you set it to the last time you changed those settings while in night mode? The default night profile could then be something sane such as minimum brightness and no bluetooth.

I'm not saying that solution doesn't have its own issues or that I'm against the ideas -- just throwing out there that the decision should be taken with care :)

**Reply by smaybius** [reply]
I'm meaning it for to work the same exact way as on regular Android and iOS, and the user can easily override the settings that airplane mode and bedtime mode change. All they really do is be a macro that bulk-changes settings when switched on, and undo their changes when switched off.
